### DISCUSSION TITLE
规则引擎sql测试成功，但是实际运行显示 undefined

### DISCUSSION THREAD

**Discussion by zhengguoxin** [emqx/emqx#5602] [Q&A]
原始数据：
{
    "test":{
        "a":1,
        "b":2,
    }
}

我想提取为：
{
"a":1,
"b":2,
}

用下面sql提取数据，测试成功，但实际运行收到的数据显示 undefined，请问该如何编写sql才可以提取到数据?

SELECT  
payload.test.a as a,
payload.test.b as b,
FROM "bridge/#"

**Comment by k32** [comment]
Hello,

I wonder if all the data sent to "bridge/#" wildcard topic matches with the expected json schema? Perhaps you need to filter messages that have the specified fields.
